### PR TITLE
RFC: Header redesign

### PR DESCRIPTION
**Pull request by bhollis**
Had this one on a branch for a long time and finally got it sorta polished up. Some controversial changes:

1. Accounts move into the menu.
3. About/Backers/Shop/Whats's New/Bug Report move into the menu.
2. Rating and wish list settings move into settings.
4. Big huge search bar.

I think if I could combine About/Backers/Shop/Whats's New/Bug Report into one item I could put it back in the bar, but I don't have a great idea for that combined page.

Also note that I'm planning on adding some pages and removing others, and finally getting around to beefing up the search bar to include saved searches, etc., which is partly why I want to give search its own space.

I could merge some of the design tweaks (nicer search bar and autocomplete) without the more disruptive changes too.

Screen Shot 2019-05-24 at 12 12 17 AM
Screen Shot 2019-05-24 at 12 12 23 AM
Screen Shot 2019-05-24 at 12 13 27 AM
Screen Shot 2019-05-24 at 12 13 59 AM

### CODE REVIEW

**Comment by bhollis** [comment]
BTW if anyone wanted to recommend icons for all the menu items that'd be cool

**Comment by inexorableAce** [comment]
I don’t think this is too controversial. Wishlists would be my only concern. That’s kinda taken off and replaced ratings for a lot of folks and it wasn’t too big in the menu anyway. Could be fine now that wishlists get saved though. 

I could run the general idea by discord if you’re super worried about it? 

Otherwise, the search bar is magnificent. That will definitely help with the saved searches and general search overhaul. 

I honestly wouldn’t bother with icons for the menu items. Nothing will be universally understood enough to remove text so we’d just be taking up more space with no real gain.

**Comment by philkernick** [comment]
Great idea.  I'd leave Accounts on the menu bar so it's really clear which account you are working on.  Think of it more as a label and less as a selector.

**Comment by bhollis** [comment]
I don't know how often people have multiple accounts or switch between them - I was thinking having it in the main menu would just centralize that sort of navigation. Also, I was mostly interested in removing the accounts menu from mobile, where it takes a lot of space.

**Comment by 48klocs** [comment]
I’m ok with moving ratings off of the header strip, but I’d like to have an element to drop a wish list in the hamburger menu. I can look into that when I get back.

**Comment by bhollis** [comment]
Why would it need to be in the hamburger menu? It seems like Settings is a more natural place for that sort of thing.

**Comment by inexorableAce** [comment]
Still think it would be cool to include 48klocs’ lists in the app to just choose from / swap between. In regards to wishlists. Dunno if that’s really relevant to this PR but I guess it could be done in conjunction with the move to settings.

**Comment by kyleshay** [comment]
> Accounts move into the menu.
- No issues with that! The hamburger menu seems like a natural place for this to live, along with 'log out'
> About/Backers/Shop/Whats's New/Bug Report move into the menu.
- I also noticed that settings can be accessed from the gear icon and the hamburger menu, maybe we'll just have the gear icon next to the search button?
- It could be helpful to distinguish it from the DIM pages (loadouts/progress/vendors/etc.), so that it seems like there are less things to try to learn about.
- We could potentially have something fixed to the bottom of the menu that is just an icon for those types of things, like using `fa-bug` to report a bug (could also update the copy on the about page to let people know how to report a bug.)
Screen Shot 2019-08-01 at 12 51 04 PM

**Comment by bhollis** [comment]
Screen Shot 2019-08-03 at 10 57 54 PM
Screen Shot 2019-08-03 at 11 00 58 PM

I reworked the About page and collapsed "Backers" and "Store" into them.

File: src/app/shell/About.tsx [hunk 1]
```diff
@@ -54,44 +83,89 @@ export default class About extends React.Component<Props> {
...
+          </div>
+          <div>
+            <h2>
+              <ExternalLink href={openCollectiveLink}>
+                <AppIcon icon={faTshirt} /> {t('Header.Shop')}
+              </ExternalLink>
+            </h2>
+            <div
+              dangerouslySetInnerHTML={{
+                __html: t('Views.Support.Teespring', { link: storeLink })
```
Comment (delphiactual) [inline src/app/shell/About.tsx]: If we are going to change the value of `Views.Support.Teespring` we might as well just change the name of the key as well `Views.Support.Store`

File: src/locale/dim.json [hunk 2]
```diff
@@ -824,11 +827,11 @@
...
     "Support": {
       "BackersDetail": "Support us with a one-time or monthly donation and help us continue our active development.",
-      "FreeToDownload": "DIM is a product that is free to download and use. The source code for DIM is open sourced, and free to anyone to enhance. You will never see an ad in DIM. That is our commitment.",
+      "FreeToDownload": "DIM is a product that is free to download and use. The source code for DIM is open source and free for anyone to enhance. You will never see an ad in DIM. That is our commitment.",
       "OpenCollective": "We are using {{link}} as a service to provide compensation to our developers for their dedication and time spent on this project.",
       "Sponsors": "Sponsors",
-      "Support": "Supporting DIM",
-      "Teespring": "We also have shirts with our new logo for sale on {{link}}"
+      "Support": "Support DIM",
+      "Teespring": "We have merch with our logo and other designs for sale on {{link}}"
```
Comment (delphiactual) [inline src/locale/dim.json]:
```suggestion
      "Store": "We have merch with our logo and other designs for sale on {{link}}"
```

**Comment by delphiactual** [comment]
The following links do not work on the `About DIM` page. These are all the main large orange text with icons, not the links included in the copy.

Support DIM:  https://localhost:8080/%3Ca%20href='https://opencollective.com/dim'%20target='_blank'%20rel='noopener%20noreferrer'%3EOpenCollective%3C/a%3E

Shop: https://localhost:8080/%3Ca%20href='https://opencollective.com/dim'%20target='_blank'%20rel='noopener%20noreferrer'%3EOpenCollective%3C/a%3E

Github: https://localhost:8080/%3Ca%20href='https://github.com/DestinyItemManager/DIM/'%20target='_blank'%20rel='noopener%20noreferrer'%3EGitHub%3C/a%3E

Join the Translation Team: https://localhost:8080/%3Ca%20href='https://crowdin.com/project/destiny-item-manager/invite?d=65a5l46565176393s2a3p403a3u22323e46383232393h4k4r443o4h3d4c333t2a3j4f453f4f3o4u643g393b343n4%27%20target=%27_blank%27%20rel=%27noopener%20noreferrer%27%3ECrowdin%3C/a%3E

**Comment by bhollis** [comment]
Thanks for the fixes!

**Comment by bhollis** [comment]
OK, let's see how people feel about this.

**Pull request merged**